src/typeguard/_checkers.py

```diff
@@ -530,6 +531,17 @@ def check_typevar(
             )
 
 
+if hasattr(typing, "Literal"):
+
+    def _is_literal_type(typ):
+        return typ is typing.Literal or typ is typing_extensions.Literal
+
+else:
+
+    def _is_literal_type(arg):
+        return typ is typing_extensions.Literal
```

Comment:
I have https://github.com/python/typing_extensions/pull/203 up currently, to provide a more generalised solution to this kind of problem. Please comment on the PR if you'd find the utility functions proposed in that PR useful! :-)

Reply:
> Hmm, I see `typing_extensions` is currently only a dependency of `typeguard` on Python <3.11. The approach I'm taking in this PR will be simpler if you can assume that `typing_extensions` is always installed. Would you be open to changing the dependencies so that `typing_extensions` is a dependency on all Python versions?

I suppose I could agree to that.

Reply:
I thought you needed `typing_extensions` to be always available for some fundamental reason, but from the looks of it, it could be easily avoided (correct me if I'm wrong).

Reply:
> I thought you needed `typing_extensions` to be always available for some fundamental reason, but from the looks of it, it could be easily avoided (correct me if I'm wrong).

Yes, looks like we can avoid it, you're right. I've updated my PR so that it no longer requires `typing_extensions` unconditionally.

In the long term, though, I think the test failures on 3.8/3.9 here may speak to a broader problem in typeguard currently. I think there may be a few other objects where you assume that typing_extensions simply reexports the object from typing on a particular Python version. In general, that's an assumption you should never make, as we could easily make a change in `typing_extensions` to backport a bugfix from CPython (which, from our perspective, wouldn't be a "breaking change") that would violate that assumption. We recently added some docs on this here: https://typing-extensions.readthedocs.io/en/latest/#runtime-use-of-types

As such, you might need to make similar changes to the one I'm making here in other places across the typeguard codebase; and you might decide at a certain point that having to guard imports of typing_extensions in `try`/`except` clauses everywhere just gets too ugly/tedious. But that's your decision, not mine :)

Reply:
> I think there may be a few other objects where you assume that typing_extensions simply reexports the object from typing on a particular Python version.

There are many occasions where a bugfix involved supporting both flavors of some types (stdlib and `typing_extensions`), but that's only done where I know `typing_extensions` has an upgraded version of some type. Perhaps it would make sense to do this pessimistically and always add all the relevant types from `typing_extensions` type to the lookup dict. If the type is just re-exported from the stdlib, then there's no harm done.

Reply:
Would you like to me to add that to this PR, or would you prefer it to be tackled in a separate PR?

Reply:
If you can do that while keeping `typing_extensions` an optional dependency on < 3.11, then sure, go ahead!

Reply:
Done!